### DISCUSSION TITLE
Robot Operating System integrate ?

### DISCUSSION THREAD

**Discussion by WhoIsMrSentry** [makerforgetech/modular-biped#57] [Ideas]
Hello Dan and community, I have been learning ROS (Robot Operating System) for a while and when it is finished, I will try to integrate the robot into ROS2 using nodes and topics. If there is anyone who has knowledge about ROS, I thought we could work together. When I look at it, the companion robot works entirely with pubsub.

**Comment by danic85** [comment]
Yes indeed. I'd be interested to see how this goes - Due to the barrier to entry for ROS (it was only ROS 1 at the time I started) I didn't build this with ROS, but I'd love to see a version that utilises it, and what advantages it brings.

Viam is another platform I'm interested in integrating, at some point :sweat_smile:

**Reply by WhoIsMrSentry** [reply]
Viam can also be very interesting, but I think ROS provides more advantages both in terms of processing and speed but there is an option: we can do it with ros2 and transfer it directly to viam (at least it is stated that way😅).

**Reply by danic85** [reply]
Viam and ROS are not mutually exclusive, you can use both in a project. 

I'd still say the barrier to entry for ROS is high enough that I wouldn't want to make it a hard dependency for this project. It would be great to see it as an optional extra however, or an alternative implementation.

**Comment by jacobsayono** [comment]
@danic85 i would love to contribute and work on ROS2 for the project. there are plenty of open-source packages that can be used to extend the scope of this robot!

@SentryCoderDev let's start by building a URDF file to describe the physical robot's joints and links. please let me know if you'd like to work on this together!

i'll use my limited knowledge of ROS2 to start developing a simple prototype architecture. i'm more than happy to learn along the way together with the community.

i can't afford a 3D printer yet so hopefully i can contribute enough within simulation for now!

**Reply by danic85** [reply]
Hi Jacob, that sounds great. It's a brilliant framework and would be good to see it implemented. Let me know if I can help with anything :crossed_fingers:

**Reply by evreidan** [reply]
I'm interested in this thread. Would like to help. My skills and knowledge are limited. So at this point will follow and learn.

**Comment by danic85** [comment]
No updates for over a year. Closing.
